api/desecapi/pdns_change_tracker.py

```diff
@@ -0,0 +1,228 @@
+        if self._rr_set_modifications.get(rr_set.domain.name, None) is None:
+            self._rr_set_additions[rr_set.domain.name] = set()
+            self._rr_set_modifications[rr_set.domain.name] = set()
+            self._rr_set_deletions[rr_set.domain.name] = set()
+
+        additions = self._rr_set_additions[rr_set.domain.name]
+        modifications = self._rr_set_modifications[rr_set.domain.name]
+        deletions = self._rr_set_deletions[rr_set.domain.name]
+
+        item = (rr_set.type, rr_set.subname)
+        if created:
+            additions.add(item)
+            assert item not in modifications
+            deletions.discard(item)
+        elif deleted:
+            if item in additions:
+                additions.remove(item)
+                modifications.discard(item)
+                # no change to deletions
+            else:
+                # item not in additions
+                modifications.discard(item)
+                deletions.add(item)
+        elif not created and not deleted:
+            # we don't care if item was created or not
+            modifications.add(item)
+            assert item not in deletions
+        else:
+            raise ValueError('An RR set cannot be created and deleted at the same time.')
+
+    def _update_domain(self, domain: Domain, created=False, deleted=False):
+        if not created and not deleted:
+            raise UnsupportedChangeException()  # TODO can this be relaxed to just no-op?
+
+        name = domain.name
+        additions = self._domain_additions
+        deletions = self._domain_deletions
+
+        if created and deleted:
+            raise ValueError('A domain set cannot be created and deleted at the same time.')
+
+        if created:
+            if name in deletions:
+                deletions.remove(name)
+            else:
+                additions.add(name)
+        elif deleted:
+            if name in additions:
+                additions.remove(name)
+            else:
+                deletions.add(name)
+
+    def _on_rr_post_save(self, signal, sender, instance: RR, created, update_fields, raw, using):
+        self._update_rr_set(instance.rrset)
+
+    def _on_rr_post_delete(self, signal, sender, instance: RR, using):
+        self._update_rr_set(instance.rrset)
+
+    def _on_rr_set_post_save(self, signal, sender, instance: RRset, created, update_fields, raw, using):
+        self._update_rr_set(instance, created=created)
+
+    def _on_rr_set_post_delete(self, signal, sender, instance: RRset, using):
+        self._update_rr_set(instance, deleted=True)
+
+    def _on_domain_post_save(self, signal, sender, instance: Domain, created, update_fields, raw, using):
+        self._update_domain(instance, created=created)
+
+    def _on_domain_post_delete(self, signal, sender, instance: Domain, using):
+        self._update_domain(instance, deleted=True)
+
+    def clear(self):
+        self._rr_set_modifications = []
```

Comment:
Why only modifications, and not also additions and deletions (of both RRset and Domain objects)? Also, I can't find where this is used.

Reply:
That's dead code